### PR TITLE
Integrate the gopath hacks

### LINKED ISSUE #145: Use GOPATH hacks to codegen anywhere.
Please see https://github.com/knative/eventing/pull/4621 or https://github.com/knative/serving/pull/10245 for an example.

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

# Changes
Integrate the gopath hacks

/kind cleanup

Fixes #145

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -18,18 +18,23 @@ set -o errexit
...
+echo "=== Update Codegen for $MODULE_NAME"
+
+group "Kubernetes Codegen"
 
 REPO_ROOT=$(dirname ${BASH_SOURCE})/..
 CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
 
+group "Knative Codegen"
+
 KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-$(cd ${REPO_ROOT}; ls -d -1 ./vendor/knative.dev/pkg 2>/dev/null || echo ../pkg)}
```
Comment (n3wscott) [inline hack/update-codegen.sh:35]: REPO_ROOT needs to be updated

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -18,18 +18,23 @@ set -o errexit
...
 export GOFLAGS=-mod=
 
-if [ -z "${GOPATH:-}" ]; then
-  export GOPATH=$(go env GOPATH)
-fi
+echo "=== Update Codegen for $MODULE_NAME"
+
+group "Kubernetes Codegen"
 
 REPO_ROOT=$(dirname ${BASH_SOURCE})/..
```
Comment (n3wscott) [inline hack/update-codegen.sh:30]: You can drop this

**Comment by arghya88** [comment]
@n3wscott Done the changes.PTAL

**Review by n3wscott — COMMENTED** [review]
/lgtm
/approve

File: hack/update-codegen.sh [hunk 3]
```diff
@@ -18,18 +18,18 @@ set -o errexit
...
+
 # If we run with -mod=vendor here, then generate-groups.sh looks for vendor files in the wrong place.
 export GOFLAGS=-mod=
 
-if [ -z "${GOPATH:-}" ]; then
-  export GOPATH=$(go env GOPATH)
-fi
+echo "=== Update Codegen for $MODULE_NAME"
+
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
```
Comment (n3wscott) [inline hack/update-codegen.sh:28]: you can drop this line,

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -18,18 +18,18 @@ set -o errexit
...
 
-REPO_ROOT=$(dirname ${BASH_SOURCE})/..
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/knative.dev/pkg 2>/dev/null || echo ../pkg)}
```
Comment (n3wscott) [inline hack/update-codegen.sh:30]: and this line.
/hold

**Comment by arghya88** [comment]
@n3wscott changes done.PTAL

**Review by n3wscott — COMMENTED** [review]
/lgtm
/approve

**Comment by arghya88** [comment]
@@chizhg can we get this in?

**Pull request merged**